style: UX Refinements on tagging components [FC-0036]

ChrisChV opened the pull request:
<!--

🌳🌳
🌳🌳🌳🌳         🌳 Note: Quince is in support. Fixes you make on master may still be needed on Quince.
    🌳🌳🌳🌳     If so, make another pull request against the open-release/quince.master branch,
🌳🌳🌳🌳         or ask in the #wg-build-test-release Slack channel if you have any questions or need help.
🌳🌳

🌴🌴🌴🌴🌴🌴     🌴 Note: the Palm release is still supported.
                Please consider whether your change should be applied to Palm as well.

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly
readable.  If you must linked information must be private (because it has secrets),
clearly label the link as private.

-->

## Description

This PR fix/adds the following:

- [X] 1. https://github.com/openedx/modular-learning/issues/118#issuecomment-1829635727: "Please change the order of the items in the unit menu to:"
- [X] 2. https://github.com/openedx/modular-learning/issues/120#issuecomment-1829706783: "Please change the order of the items in the component menu to:"
Manage Access
- [x] 3. https://github.com/openedx/modular-learning/issues/120#issuecomment-1829706783: "There is a strange 'jump' when I first expand the "Tags" sidebar component. The area containing the "Tags" heading, tag count, and up/down arrow seems to shift down a few pixels when it is first clicked"
- [x] 4. https://github.com/openedx/modular-learning/issues/120#issuecomment-1829706783: This section also shows a dotted outline after it is selected
- [x] 5. https://github.com/openedx/modular-learning/issues/120#issuecomment-1829706783: "If a tag cannot be expanded (i.e. leaf tags), it should not have a hover state"
- [x] 7. Change the default URL for the "Taxonomies" tab to go to /taxonomies/ directly, not /taxonomy-list/

## Supporting information

Github issue: https://github.com/openedx/modular-learning/issues/157
Internal ticket: [FAL-3537](https://tasks.opencraft.com/browse/FAL-3537)

## Testing instructions

- Enable `contentstore.enable_copy_paste_units` and `new_studio_mfe.use_tagging_taxonomy_list_page` flags.
- Before to all you need to run the [taxonomy-sample-data](https://github.com/open-craft/taxonomy-sample-data).

**"Please change the order of the items in the unit menu to:"**

- Go to a [course](http://localhost:18010/course/course-v1:SampleTaxonomyOrg2+STC1+2023_1), expand an unit and verify the new order of the menu.

**"Please change the order of the items in the component menu to:"**

- Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify the new order of the menu

**There is a strange 'jump' when I first expand the "Tags" sidebar component. The area containing the "Tags" heading, tag count, and up/down arrow seems to shift down a few pixels when it is first clicked**

- Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify that the 'jump' is gone when you expand de 'Tags' component.

**This section also shows a dotted outline after it is selected**

- Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify the dotted outline is gone when you click on 'Tags' componene. But when you use the keyboard should appear.

**If a tag cannot be expanded (i.e. leaf tags), it should not have a hover state**

- Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify the change on the 'Tags' component

**Change the default URL for the "Taxonomies" tab to go to /taxonomies/ directly, not /taxonomy-list/**

- Go to [studio home](http://localhost:18010/home/) and click on the 'Taxonomies' tab. Verify the new URL.

## Other information

**6. Slack: Refresh the count of tags on the unit/outline page(s) after the user makes edits in the tag drawer**

This issue has not been fixed since https://github.com/openedx/frontend-app-course-authoring/pull/704 is still open

ChrisChV committed:
style: Update order of unit/component menus

openedx-webhooks commented:
Thanks for the pull request, @ChrisChV! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

ChrisChV committed:
style: Nits and updates on unit tag list component

ChrisChV committed:
feat: Update MFE taxonomy list page URL

ChrisChV committed:
Merge branch 'master' into chris/FAL-3537-ux-refinements

ChrisChV committed:
test: Fix tests

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at [pr-33884-139931.staging.do.opencraft.hosting](https://pr-33884-139931.staging.do.opencraft.hosting)
Sandbox Studio is available at [studio.pr-33884-139931.staging.do.opencraft.hosting](https://studio.pr-33884-139931.staging.do.opencraft.hosting)

Instance config : [https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5696107567-config.yml](https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5696107567-config.yml)
Instance requirements : [https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5696107567-requirements.txt](https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5696107567-requirements.txt)

yusuf-musleh reviewed (approved):
:+1: @ChrisChV I checked all the cases, things seem to be working as expected, however I had doubts/issues with two of them:


> **There is a strange 'jump' when I first expand the "Tags" sidebar component. The area containing the "Tags" heading, tag count, and up/down arrow seems to shift down a few pixels when it is first clicked**
> 
> * Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify that the 'jump' is gone when you expand de 'Tags' component.

The jump is now gone when clicking to expand the `Tags` component. However when you tab on the page and the focus becomes on the 'Tags' header, things get shifted a few pixels as well, not sure if its the same "jump" or not.
 
> **If a tag cannot be expanded (i.e. leaf tags), it should not have a hover state**
> 
> * Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify the change on the 'Tags' component

This is now fixed, however when clicking on a tag to expand/close, it remains blue, until I click elsewhere, is that intended behavior?

- [X] I tested this: (I checked all the cases to confirm they are working, I have doubts on 2 of them though)
- [X] I read through the code
- [X] I checked for accessibility issues
- [ ] ~Includes documentation~
- [ ] ~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~

ChrisChV committed:
style: Fix style issue with Tag border

ChrisChV commented:
> > **If a tag cannot be expanded (i.e. leaf tags), it should not have a hover state**
> > 
> > * Go to a [unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg2+STC1+2023_1+type@vertical+block@7f47fe2dbcaf47c5a071671c741fe1ab) and verify the change on the 'Tags' component
> 
> This is now fixed, however when clicking on a tag to expand/close, it remains blue, until I click elsewhere, is that intended behavior?

I think this is a question for @ali-hugo 

> The jump is now gone when clicking to expand the Tags component. However when you tab on the page and the focus becomes on the 'Tags' header, things get shifted a few pixels as well, not sure if its the same "jump" or not.

@yusuf-musleh I fixed that here https://github.com/openedx/edx-platform/pull/33884/commits/c2c4e706002e5a4404f726b97ad6b7756fbc95f5

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at [pr-33884-139931.staging.do.opencraft.hosting](https://pr-33884-139931.staging.do.opencraft.hosting)
Sandbox Studio is available at [studio.pr-33884-139931.staging.do.opencraft.hosting](https://studio.pr-33884-139931.staging.do.opencraft.hosting)

Instance config : [https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5707998831-config.yml](https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5707998831-config.yml)
Instance requirements : [https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5707998831-requirements.txt](https://grove-stage-build-logs.nyc3.digitaloceanspaces.com/34602668-5707998831-requirements.txt)

bradenmacdonald commented on cms/static/sass/views/_container.scss:
```diff
@@ -277,8 +278,8 @@
-        .wrapper-tag-header:focus {
-          border: 1px dotted gray;
+        .wrapper-tag-header:focus-visible {
+          border-color: gray;
```
Is there an SCSS variable we can use like `$gray` or `$border-color`? Just wondering.

ChrisChV replied:
Updated https://github.com/openedx/edx-platform/pull/33884/commits/9f103537c583dc6924f8c9178a6b5f626cded672

bradenmacdonald commented:
>> This is now fixed, however when clicking on a tag to expand/close, it remains blue, until I click elsewhere, is that intended behavior?
> I think this is a question for @ali-hugo

She is on holiday now or starting soon so I'm just going to make the call: I think it's strange how it stays blue now. Please remove the `:focus` so that it's only blue on hover, and it doesn't stay blue after you click.

(But as before, make sure there is a visual indication when it is focused by keyboard. I think that should still happen "automatically" but if it's not, you'll need to add a `:focus-visible` style to this one as well. But it should be a dotted rectangle outline like the others - turning a link from black to blue is too subtle and not appropriate for a keyboard focus indicator.)

bradenmacdonald reviewed (commented):
Looks great! Please just fix the issue @yusuf-musleh noticed and this will be good to go.

ChrisChV committed:
style: Nits on container styles

ChrisChV committed:
Merge branch 'master' into chris/FAL-3537-ux-refinements

ChrisChV commented:
@bradenmacdonald This is ready for merge

bradenmacdonald reviewed (approved):
:+1:

- [x] I tested this: in Studio
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation: n/a

openedx-webhooks commented:
@ChrisChV 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/33884&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2023-12-05+16:17&entry.1917517419=2023-12-13+19:28&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
